### PR TITLE
OCPBUGS-33758: Add and update CNO runbooks for ovnk-ic

### PR DESCRIPTION
**Pull request by jcaamano**
ovnk-ic introduced new components and new corresponding alerts as well as new ways of interpreting existing alerts. Alerts that are no longer current with ovnk-ic are kept around as are still serviceable on old releases although clarifying notes will are added to them.

This PR is a minimal approach to bring the alerts up to date with respect to ovnk-ic as it has been long overdue. While procuring not to strip the alerts of useful information that was already there for the equivalent non ovnk-ic version of the alerts, further improvements should be scoped on a separate PR.

This is done in reference to the corresponding CNO PR https://github.com/openshift/cluster-network-operator/pull/2406

This effort replaces https://github.com/openshift/runbooks/pull/136

### CODE REVIEW

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+You may need to use `--previous` command with `oc logs` command to get the logs
+of the previous execution run of a container. Pay close attention to any log
+output starting with "E" for Error.
+
+## Mitigation
+
+The appropriate mitigation will be very different depending on the cause of the
+error discovered in the diagnosis. Investigate the issue using the steps
+outlined in diagnosis and contact the incident response team in your
+organisation if fixing the issue is not apparent.
```
Comment (jcaamano) [inline alerts/cluster-network-operator/NoRunningOvnControlPlane.md:86]: We probably should be saying this in all alerts instead of referring to slack

Comment (jcaamano) [reply alerts/cluster-network-operator/NoRunningOvnControlPlane.md:86]: done

File: alerts/cluster-network-operator/NoOvnClusterManagerLeader.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+why the election failed.
+
+### If all the ovnkube-control-plane pods are not running
+
+Check the status of the ovnkube-control-plane pods, and follow the [Pod
+lifecycle][Pod lifecycle] to see what is blocking the pods to be running.
+
+### If all the ovnkube-control-plane pods are running
+
+Follow the steps above: [OVN-Kubernetes master
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NoOvnClusterManagerLeader.md:73]: OVN-Kubernetes control plane pods*

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# NoRunningOvnControlPlane
+
+## Meaning
+
+This alert is triggered when there are no OVN-Kubernetes control plane pods
+[Running][PodRunning] for `10m`.
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NoRunningOvnControlPlane.md:6]: for `5m`

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+## Impact
+
+When OVN-Kubernetes control plane is not running, networking control plane is
+degraded and networking configuration updates applied to the cluster will not be
+implemented. Existing workloads may continue to have connectivity but any
+additional workloads may not be provisioned successfully. Network policy changes
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NoRunningOvnControlPlane.md:13]: This is outdated now due to IC.
what about:
```
When OVN-Kubernetes control plane is not running, networking control plane is
degraded. A subset of networking control plane functionality is degraded. This includes, but not limited to the following networking control plane functionality:
* Node management including node subnet allocation/de-allocation
* DNS name resolver functionality for EgressFirewall
* EgressService node allocation/re-allocation
* Secondary networks IPAM
* EgressIP assignment/re-assignment and health checks
* OVN zone management
```

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+### Fix alerts before continuing
+
+Resolve any alerts that may cause this alert to fire: [Alert
+hierarchy](./hierarchy/alerts-hierarchy.svg)
+
+## Diagnosis
+
+### Control plane issue
+
+This can occur multiple control plane nodes are powered off or are unable to
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NoRunningOvnControlPlane.md:25]: .. occur *when* multiple..

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+connect each other via the network. Check that all control plane nodes are
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NoRunningOvnControlPlane.md:26]: ..connect *with* each other..

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+powered and that network connections between each machine are functional
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NoRunningOvnControlPlane.md:27]: ..powered *on* and..

..are functional*:*

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+```shell
+oc get node -l node-role.kubernetes.io/master=""
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NoRunningOvnControlPlane.md:29]: nit: node-role.kubernetes.io/control-plane

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+### Cluster network operator
+
+Cluster Network operator (CNO) manages the CRUD operations for OVN-Kubernetes
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NoRunningOvnControlPlane.md:34]: Cluster *n*etwork operator

File: alerts/cluster-network-operator/NorthboundStaleAlert.md [hunk 10]
```diff
@@ -1,102 +1,106 @@
...
 
 ## Meaning
 
-The alert `NorthboundStale` will be triggered if the ovn-kube master process is
-not functioning, if the northbound database is not functioning, or if
-connectivity between the ovnkube master and the database is broken.
-
-This alert will be triggered if `NoRunningOvnMaster` or if `NoOvnMasterLeader`
-is firing, so check those alerts before proceeding here.
+This alert is triggered when ovnkube-controller or northbound database processes
```
Comment (martinkennelly) [inline alerts/cluster-network-operator/NorthboundStaleAlert.md:5]: ..specific *availability* domain..

File: alerts/cluster-network-operator/NoRunningOvnControlPlane.md (hunk 4 above)
Comment (jcaamano) [reply alerts/cluster-network-operator/NoRunningOvnControlPlane.md:13]: I was being vague on purpose but I will try a version of what you are proposing

**Pull request merged**